File: rsuiteextensionscore/src/main/java/org/researchsuite/rsuiteextensionscore/RSLoginStep.java
package org.researchsuite.rsuiteextensionscore;

import org.researchstack.backbone.step.Step;

/**
 * Created by jameskizer on 3/13/18.
 */

public class RSLoginStep extends Step {

    public String getLogInButtonTitle() {
        return logInButtonTitle;
    }

    public String getForgotPasswordButtonTitle() {
        return forgotPasswordButtonTitle;
    }

    public String getIdentityFieldName() {
        return identityFieldName;
    }

    public String getPasswordFieldName() {
        return passwordFieldName;
    }

    private String logInButtonTitle = "Log In";

    public void setLogInButtonTitle(String logInButtonTitle) {
        this.logInButtonTitle = logInButtonTitle;
    }

    public void setForgotPasswordButtonTitle(String forgotPasswordButtonTitle) {
        this.forgotPasswordButtonTitle = forgotPasswordButtonTitle;
    }

    public void setIdentityFieldName(String identityFieldName) {
        this.identityFieldName = identityFieldName;
    }

    public void setPasswordFieldName(String passwordFieldName) {
        this.passwordFieldName = passwordFieldName;
    }

    private String forgotPasswordButtonTitle;
    private String identityFieldName = "Username";
    private String passwordFieldName = "Password";

    public RSLoginStep(String identifier, String title, String text, Class logInLayoutClass) {
        super(identifier, title);
        this.setText(text);
        this.setStepLayoutClass(logInLayoutClass);
    }

    public RSLoginStep(String identifier, String title, String text) {
        super(identifier, title);
        this.setText(text);
        this.setStepLayoutClass(RSLoginStepLayout.class);
    }

}

File: rsuiteextensionscore/src/main/java/org/researchsuite/rsuiteextensionscore/RSLoginStep.kt
package org.researchsuite.rsuiteextensionscore

import android.text.InputType
import org.researchstack.backbone.step.Step

open class RSLoginStep @JvmOverloads constructor(
        identifier: String,
        text: String?,
        title: String?,
        logInLayoutClass: Class<*>,
        val loginButtonTitle: String = "Login",
        val forgotPasswordButtonTitle: String? = null,
        val identityFieldName: String = "Username",
        val identityFieldInputType: Int = InputType.TYPE_CLASS_TEXT,
        val showIdentityField: Boolean = true,
        val passwordFieldName: String = "Password",
        val passwordFieldInputType: Int = (InputType.TYPE_CLASS_TEXT and InputType.TYPE_TEXT_VARIATION_PASSWORD),
        val showPasswordField: Boolean = true
): Step(identifier) {

    init {
        this.text = text
        this.title = title
        this.stepLayoutClass = logInLayoutClass
    }

}